ui-discover/src/main/java/app/tivi/home/discover/Discover.kt

```diff
@@ -0,0 +1,424 @@
+                                        showId = nextEpisodeToWatch.show.id,
+                                        episodeId = nextEpisodeToWatch.episode.id
+                                    )
+                                )
+                            }
+                        )
+                    }
+
+                    item { Spacer(Modifier.preferredHeight(16.dp)) }
+                }
+
+                item {
+                    CarouselWithHeader(
+                        items = state.trendingItems,
+                        title = stringResource(R.string.discover_trending_title),
+                        refreshing = state.trendingRefreshing,
+                        onItemClick = { actioner(OpenShowDetails(it.id)) },
+                        onMoreClick = { actioner(OpenTrendingShows) }
+                    )
+                }
+
+                item {
+                    CarouselWithHeader(
+                        items = state.recommendedItems,
+                        title = stringResource(R.string.discover_recommended_title),
+                        refreshing = state.recommendedRefreshing,
+                        onItemClick = { actioner(OpenShowDetails(it.id)) },
+                        onMoreClick = { actioner(OpenRecommendedShows) }
+                    )
+                }
+
+                item {
+                    CarouselWithHeader(
+                        items = state.popularItems,
+                        title = stringResource(R.string.discover_popular_title),
+                        refreshing = state.popularRefreshing,
+                        onItemClick = { actioner(OpenShowDetails(it.id)) },
+                        onMoreClick = { actioner(OpenPopularShows) }
+                    )
+                }
+
+                item { Spacer(Modifier.preferredHeight(16.dp)) }
+            }
+
+            DiscoverAppBar(
+                loggedIn = state.authState == TraktAuthState.LOGGED_IN,
+                user = state.user,
+                refreshing = state.refreshing,
+                onRefreshActionClick = { actioner(RefreshAction) },
+                onUserActionClick = { actioner(OpenUserDetails) },
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .onSizeChanged { appBarHeight = it.height }
+            )
+        }
+    }
+}
+
+@Composable
+private fun NextEpisodeToWatch(
+    show: TiviShow,
+    poster: TmdbImageEntity?,
+    season: Season,
+    episode: Episode,
+    modifier: Modifier = Modifier,
+) {
+    Surface(modifier) {
+        Row(Modifier.padding(horizontal = 16.dp, vertical = 8.dp)) {
+            if (poster != null) {
+                PosterCard(
+                    show = show,
+                    poster = poster,
+                    modifier = Modifier.preferredWidth(64.dp).aspectRatio(2 / 3f)
+                )
+            }
+
+            Spacer(Modifier.preferredWidth(16.dp))
```

Comment:
Should this go inside the above `if` i.e. don't you only want this is there is a poster?

Reply:
Good catch